src/Supervisor.cpp

```diff
@@ -1221,4 +1221,63 @@ ZunResult Supervisor::FadeOutMusic(f32 fadeOutSeconds)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+#pragma optimize("s", on)
+// this is for rebinding keys
+u8 *th06::Controller::GetControllerState()
+{
+    static u32 controllerData[32];
+    memset(&controllerData, 0, sizeof(controllerData));
+    if (g_Supervisor.controller == NULL)
+    {
+        // TODO: not tested
+        JOYINFOEX joyinfo;
+        memset(&joyinfo, 0, sizeof(JOYINFOEX));
+        joyinfo.dwSize = sizeof(JOYINFOEX);
+        joyinfo.dwFlags = JOY_RETURNALL;
+        MMRESULT MVar1 = joyGetPosEx(0, &joyinfo);
+        if (MVar1 == 0)
```

Comment:
Usually, when something like this happens, it means that the function its called inside the if, so this would be
```c
if (joyGetPosEx(0, &joyinfo) == JOYERR_NOERROR)
```